### PR TITLE
driver: usb_serial: esp32c3: fix USB port behavior

### LINKED ISSUE #52605: esp32-usb-serial tx-complete interrupt not working in interrupt mode on esp32c3
**Describe the bug**
Hi! I'm running into some issue when trying to use the `usb_serial` device on an ESP32C3 system. When running the `shell_module` sample, I can tell that the data is received correctly, but the output buffer is only flushed as a reaction to a receive event. For example I can type a command, but the echo is delayed and the output is not printed until something else is sent on the serial port. The behavior seems to indicate that `serial_esp32_usb_isr` is not being called as it should when the transmit buffer is empty, hence stopping the application from flushing its transmit queue.

The driver works correctly in polling mode (`CONFIG_SHELL_BACKEND_SERIAL_INTERRUPT_DRIVEN=n`).

Additionally, the USB debug interface fails entirely when building the wifi sample (that is, does not enumerate, errors in dmesg, have to reset it in bootloader mode to flash it again). Could this be some interrupt management issue?

Disclaimer: I'm actually using a Seeed XIAO-ESP32-C3 board, not the official `esp32c3_devkitm`, but I think the problem should be the same since the device is builtin.

**To Reproduce**
1. patch `boards/riscv/esp32c3_devkitm/esp32c3_devkitm.dts`
```patch
-               zephyr,shell-uart = &uart0;
+               zephyr,shell-uart = &usb_serial;
```
2. build and flash the `shell_module` sample
``` 
west build -p -b esp32c3_devkitm samples/subsys/shell/shell_module
west flash
```
3. access the shell interface (minicom, west espressif monitor...), send some command, output is not flushing
4. build and flash with `west build -p -b esp32c3_devkitm samples/subsys/shell/shell_module -DCONFIG_SHELL_BACKEND_SERIAL_INTERRUPT_DRIVEN=y`, the shell works fine

**Expected behavior**
The USB serial works as expected.

**Impact**
Annoyance

**Environment (please complete the following information):**
 - Linux, Zephyr SDK 0.15.2, main (55843e7d77), Seeed XIAO-ESP32-C3

### PR DESCRIPTION
**Pull request by sylvioalves**
- Fixes missing poll out events when UART interrupt is enabled
- Fixes blocking USB interface during board start up.

Fixes #52605

### CODE REVIEW

**Comment by sylvioalves** [comment]
@fabiobaltieri @cfriedt , would you check this PR in the icev_wireless board or any other you repoduce the issue?

**Comment by fabiobaltieri** [comment]
Tested on `xiao_esp32c3` with `samples/subsys/shell/shell_module` and `samples/bluetooth/peripheral_hids`, seems to work fine. Thanks!

File: drivers/serial/serial_esp32_usb.c [hunk 1]
```diff
@@ -53,14 +58,17 @@ static void serial_esp32_usb_poll_out(const struct device *dev, unsigned char c)
 	ARG_UNUSED(dev);
 
 	/* Wait for space in FIFO */
-	while (usb_serial_jtag_ll_txfifo_writable() == 0) {
-		;
+	while (!usb_serial_jtag_ll_txfifo_writable() &&
+		s_usbserial_timeout < (USBSERIAL_TIMEOUT_MAX_US / 100)) {
+		esp_rom_delay_us(100);
```
Comment (fabiobaltieri) [inline drivers/serial/serial_esp32_usb.c:63]: Would it make sense to use one of the Zephyr APIs for the delay here?

Comment (sylvioalves) [reply drivers/serial/serial_esp32_usb.c]: done!

**Review by fabiobaltieri — APPROVED** [review]

**Comment by cfriedt** [comment]
> @fabiobaltieri @cfriedt , would you check this PR in the icev_wireless board or any other you can reproduce the issue?

Will do!

**Review by carlescufi — APPROVED** [review]

**Comment by cfriedt** [comment]
Works for me too 🙌 Thanks @sylvioalves !

**Pull request merged**